packages/features/ee/organizations/components/CreateANewOrganizationForm.tsx

```diff
@@ -0,0 +1,293 @@
+import { signIn } from "next-auth/react";
+import { useRouter } from "next/router";
+import type { Dispatch, SetStateAction } from "react";
+import { useState } from "react";
+import { Controller, useForm } from "react-hook-form";
+
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import slugify from "@calcom/lib/slugify";
+import { telemetryEventTypes, useTelemetry } from "@calcom/lib/telemetry";
+import { trpc } from "@calcom/trpc/react";
+import {
+  Button,
+  Form,
+  TextField,
+  Alert,
+  Dialog,
+  DialogClose,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  Label,
+} from "@calcom/ui";
+import { ArrowRight, Info } from "@calcom/ui/components/icon";
+
+function extractDomainFromEmail(email: string) {
+  let out = "";
+  try {
+    const match = email.match(/^(?:.*?:\/\/)?.*?(?<root>[\w\-]*(?:\.\w{2,}|\.\w{2,}\.\w{2}))(?:[\/?#:]|$)/);
+    out = (match && match.groups?.root) ?? "";
+  } catch (ignore) {}
+  return out.split(".")[0];
+}
+
+export const VerifyCodeDialog = ({
+  isOpenDialog,
+  setIsOpenDialog,
+  email,
+  onSuccess,
+}: {
+  isOpenDialog: boolean;
+  setIsOpenDialog: Dispatch<SetStateAction<boolean>>;
+  email: string;
+  onSuccess: (isVerified: boolean) => void;
+}) => {
+  const { t } = useLocale();
+  // Not using the mutation isLoading flag because after verifying we submit the underlying org creation form
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState("");
+
+  const [inputCode, setInputCode] = useState("");
+
+  const verifyCodeMutation = trpc.viewer.organizations.verifyCode.useMutation({
+    onSuccess: (data) => {
+      setIsLoading(false);
+      onSuccess(data);
+    },
+    onError: (err) => {
+      setIsLoading(false);
+      if (err.message === "invalid_code") {
+        setError("The code provided is not valid, try again");
+      }
+    },
+  });
+
+  return (
+    <Dialog open={isOpenDialog} onOpenChange={setIsOpenDialog}>
+      <DialogContent className="sm:max-w-md">
+        <div className="flex flex-row">
+          <div className="w-full">
+            <DialogHeader title="Verify your email" subtitle={`Enter the 6 digit code we sent to ${email}`} />
```

Comment:
i18n string required

Reply:
Resolved in the [correct PR](https://github.com/calcom/cal.com/pull/9184) for these changes, thanks!